x-pack/plugins/ml/public/settings/calendar/edit/new_calendar.js

```diff
@@ -0,0 +1,310 @@
+
+    try {
+      await ml.addCalendar(calendar);
+      window.location = `${chrome.getBasePath()}/app/ml#/settings/calendars_list`;
+    } catch (error) {
+      console.log('Error saving calendar', error);
+      this.setState({ saving: false });
+      toastNotifications.addDanger(`An error occurred creating calendar: ${calendar.calendarId}`);
+    }
+  }
+
+  onEdit = async () => {
+    const calendar = this.setUpCalendarForApi();
+    this.setState({ saving: true });
+
+    try {
+      await ml.updateCalendar(calendar);
+      window.location = `${chrome.getBasePath()}/app/ml#/settings/calendars_list`;
+    } catch (error) {
+      console.log('Error saving calendar', error);
+      this.setState({ saving: false });
+      toastNotifications.addDanger(`An error occurred saving calendar: ${calendar.calendarId}. Try refreshing the page.`);
+    }
+  }
+
+  setUpCalendarForApi = () => {
+    const {
+      formCalendarId,
+      description,
+      events,
+      selectedGroupOptions,
+      selectedJobOptions,
+    } = this.state;
+
+    const jobIds = selectedJobOptions.map((option) => option.label);
+    const groupIds = selectedGroupOptions.map((option) => option.label);
+
+    // set up calendar
+    const calendar = {
+      calendarId: formCalendarId,
+      description,
+      events,
+      job_ids: [...jobIds, ...groupIds]
+    };
+
+    return calendar;
+  }
+
+  onCreateGroupOption = (newGroup) => {
+    const newOption = {
+      label: newGroup,
+    };
+    // Select the option.
+    this.setState(prevState => ({
+      selectedGroupOptions: prevState.selectedGroupOptions.concat(newOption),
+    }));
+  };
+
+  onJobSelection = (selectedJobOptions) => {
+    this.setState({
+      selectedJobOptions,
+    });
+  };
+
+  onGroupSelection = (selectedGroupOptions) => {
+    this.setState({
+      selectedGroupOptions,
+    });
+  };
+
+  onCalendarIdChange = (e) => {
+    const isValid = validateCalendarId(e.target.value);
+
+    this.setState({
+      formCalendarId: e.target.value,
+      isNewCalendarIdValid: isValid
+    });
+  };
+
+  onDescriptionChange = (e) => {
+    this.setState({
+      description: e.target.value,
+    });
+  };
+
+  showImportModal = () => {
+    this.setState(prevState => ({
+      isImportModalVisible: !prevState.isImportModalVisible,
+    }));
+  }
+
+  closeImportModal = () => {
+    this.setState({
+      isImportModalVisible: false,
+    });
+  }
+
+  onEventDelete = (eventId) => {
+    this.setState(prevState => ({
+      events: prevState.events.filter(event => event.event_id !== eventId)
+    }));
+  }
+
+  closeNewEventModal = () => {
+    this.setState({ isNewEventModalVisible: false });
+  }
+
+  showNewEventModal = () => {
+    this.setState({ isNewEventModalVisible: true });
+  }
+
+  addEvent = (event) => {
+    this.setState(prevState => ({
+      events: [...prevState.events, event],
+      isNewEventModalVisible: false
+    }));
+  }
+
+  addImportedEvents = (events) => {
```

Comment:
In tests I had a few problems importing events from an ics file. Once I saw a 'ParseError' in the JavaScript console, but no error in the UI, and secondly, when trying to import past events no error was seen but no events seemed to be imported. This area probably needs more testing.

Reply:
Heya @peteharverson - I fixed the problem of not being able to import past events but I haven't been able to reproduce the issue where you see the 'ParseError' in the console but no error on the modal. 🤔 
Could you give me more context or share the file if you get a chance? I'll test this more as well on my end.

Reply:
I can't reproduce this with your latest edits, and see `ParseError` for example when trying to import e.g. a text file